packages/mui-system/src/Container/createContainer.d.ts

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { CreateStyledComponent } from '@mui/styled-engine';
+import { MUIStyledCommonProps } from '../createStyled';
+import Container, { ContainerProps } from './Container';
+import { Theme as DefaultTheme } from '../createTheme';
+
+export default function createContainer<Theme extends object = DefaultTheme>(options?: {
+  createStyledComponent?: CreateStyledComponent<
```

Comment:
@siriwatknp did you manage to successfully type this param so far? This is the closest I can get, but there are still some tss issues. I've tried converting the `createContainer.js` to `createContainer.tsx` and use `typeof defaultCreateStyledComponent`, but then when it is used with other theme it fails too (for example in Joy). There are some function overloads happening there that make my mind explode :D 

I almost lean towards providing more params to the function, like `overridesResolver`, `componentName` and `styled` just for the sake of having correct types.